lib/active_merchant/billing/gateways/mundipagg.rb

```diff
@@ -288,14 +304,26 @@ def commit(action, parameters, auth = nil)
         )
       end
 
-      def success_from(response)
-        %w[pending paid processing canceled active].include? response['status']
+      def extract_charge_object_from_response(response, url)
+        orders_endpoint?(url) ? charge_object(response) : response
+      end
+
+      def charge_object(response)
+        response.try(:[], 'charges').try(:[], 0)
```

Comment:
The use of `try` here is misleading. If for some reason we cannot find the `charges` array in the response and this method returns `nil`, other methods called in `def commit` will break. This leads to a bad situation where the transaction has already been processed by the gateway but we blow up on the way to returning a response to the merchant.

This shouldn't ever happen as long as the gateway is returning responses the way we expect. But I am open to suggestions about how to protect ourselves in this scenario.

Reply:
The only place to consider guarding against the scenario where this function returns `nil` is where the returned value itself is used. So, in `message_from` you could first `return unless charge`, and in the case it is `nil` it should avoid runtime errors.

Reply:
This makes sense. I resisted doing this because the end result would be to return a failure response without a message. I suppose that is preferable to a runtime error, but it still feels icky because it's possible that the transaction was successful, but then because we couldn't parse the response, `success_from` is nil, `message_from` is nil, and `error_code` is nil.

I will change it to include the guard clause like you describe.